hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -242,28 +241,39 @@ protected void getAllInternal(Set<? extends K> keys, Collection<Data> dataKeys,
             return;
         }
 
+        Map<Object, Long> reservations = createHashMap(ncKeys.size());
+        Map<Data, Object> reverseKeyMap = null;
         if (!serializeKeys) {
-            keyMap = toDataKeys(ncKeys, dataKeys);
+            reverseKeyMap = createHashMap(ncKeys.size());
+            toDataKeysWithReservations(ncKeys, dataKeys, reservations, reverseKeyMap);
+        } else {
+            createNearCacheReservations(dataKeys, reservations);
```

Comment:
If we are using serialized keys, we just get the reservations from the serialized keys.